pennylane/transforms/compile.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for a compilation transform."""
+
+from inspect import getmembers, isclass
+import pennylane.templates as templates
+
+# Get a list of the names of existing templates in PennyLane
+template_list = [cls[0] for cls in getmembers(templates, isclass)]
```

Comment:
This is used to make sure that templates are expanded into elementary gates in situations when a user does not specify a basis set; this is because
 - If we just run a blanket `expand` at depth 1, non-template gates will also be expanded, which may not be desired
 - If we don't expand the templates, the gates within will not be included in the optimization because the template is counted as an operation in the tape, and its name is the template name   

However it's not a very satisfying solution.

Reply:
tbh I can’t think of a satisfying solution here either 🤔